### DISCUSSION TITLE
What's the best way to handle assignment of null const char* pointers?

### DISCUSSION THREAD

**Discussion by prithvisb** [nlohmann/json#4003] [Q&A]
I can assign a nullptr directly to a json object

json j["foo"] = nullptr;

And the output is 
{"foo": null}

However, I can't do this with a const char* that is null

```
const char* bar = nullptr;
json j["foo"] = bar;
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid

```
I understand this is because you can't pass nullptr to a C++ basic string constructor. But apart from a check at assignment 
```

json j;
const char* bar;
if(bar)
  j["foo"] = bar;
else
  j["foo"] = nullptr;
```

Is there a better way?

**Comment by nlohmann** [comment]
Storing `(const char*)nullptr` is not possible:

- A `nullptr` of type `nullptr_t` is used to store JSON's `null` value.
- `const char*` values are forwarded to the `std::string` constructor which does not allow `NULL` values.

So I'm afraid there is no other way.
